src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -168,6 +170,38 @@ internal void LoadContext(ExecutionContext context, bool checkReturnValue)
             LoadContext(context);
         }
 
+        public ExecutionContext LoadContract(ContractState contract, string method, CallFlags callFlags, bool packParameters = false)
+        {
+            ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod(method);
+            if (md is null) return null;
+
+            ExecutionContext context = LoadScript(contract.Script, callFlags, md.Offset);
+
+            if (NativeContract.IsNative(contract.ScriptHash))
+            {
+                if (packParameters)
+                {
+                    using ScriptBuilder sb = new ScriptBuilder();
+                    sb.Emit(OpCode.DEPTH, OpCode.PACK);
+                    sb.EmitPush(md.Name);
```

Comment:
Without pack we should push the name too

Reply:
The name should be pushed with agrs.